src/transformers/models/mamba/modeling_mamba.py

```diff
@@ -640,14 +672,31 @@ def prepare_inputs_for_generation(
         cache_params: Optional[MambaCache] = None,
         **kwargs,
     ):
+        use_cache = use_cache if use_cache is not None else (self.config.use_cache if not self.training else False)
```

Comment:
a lot of this is already done in the model's forward, let's not duplicate if possible